Source/Plugins/Sampling/SamplingFilters/SampleSurfaceMeshSpecifiedPoints.cpp

```diff
@@ -0,0 +1,280 @@
+  setOutputFilePath(reader->readString("OutputFilePath", getOutputFilePath()));
+  reader->closeFilterGroup();
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+int SampleSurfaceMeshSpecifiedPoints::writeFilterParameters(AbstractFilterParametersWriter* writer, int index)
+{
+  SampleSurfaceMesh::writeFilterParameters(writer, index);
+  writer->openFilterGroup(this, index);
+  DREAM3D_FILTER_WRITE_PARAMETER(InputFilePath)
+  DREAM3D_FILTER_WRITE_PARAMETER(OutputFilePath)
+  writer->closeFilterGroup();
+  return ++index; // we want to return the next index that was just written to
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void SampleSurfaceMeshSpecifiedPoints::updateVertexInstancePointers()
+{
+	setErrorCondition(0);
+	if (NULL != m_FeatureIdsPtr.lock().get()) /* Validate the Weak Pointer wraps a non-NULL pointer to a DataArray<T> object */
+	{
+		m_FeatureIds = m_FeatureIdsPtr.lock()->getPointer(0);
+	} /* Now assign the raw pointer to data from the DataArray<T> object */
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void SampleSurfaceMeshSpecifiedPoints::dataCheck()
+{
+  setErrorCondition(0);
+  DataArrayPath tempPath;
+
+  if (true == m_InputFilePath.isEmpty())
+  {
+	  QString ss = QObject::tr("The Input file name must be set before executing this filter.");
+	  setErrorCondition(-1);
+	  notifyErrorMessage(getHumanLabel(), ss, -1);
+  }
+  if (true == m_OutputFilePath.isEmpty())
+  {
+	  QString ss = QObject::tr("The Output file name must be set before executing this filter.");
+	  setErrorCondition(-1);
+	  notifyErrorMessage(getHumanLabel(), ss, -1);
+  }
+
+  DataContainer::Pointer v = getDataContainerArray()->createNonPrereqDataContainer<AbstractFilter>(this, "SpecifiedPoints");
+  if (getErrorCondition() < 0) { return; }
+
+  VertexGeom::Pointer vertices = VertexGeom::CreateGeometry(1, DREAM3D::Geometry::VertexGeometry);
+  v->setGeometry(vertices);
+
+  QVector<size_t> tDims(1, 0);
+  AttributeMatrix::Pointer vertexAttrMat = v->createNonPrereqAttributeMatrix<AbstractFilter>(this, "SpecifiedPointsData", tDims, DREAM3D::AttributeMatrixType::Vertex);
+  if (getErrorCondition() < 0 || NULL == vertexAttrMat.get()) { return; }
+
+  QVector<size_t> cDims(1, 1);
+  tempPath.update("SpecifiedPoints", "SpecifiedPointsData", "FeatureIds");
+  m_FeatureIdsPtr = getDataContainerArray()->createNonPrereqArrayFromPath<DataArray<int32_t>, AbstractFilter, int32_t>(this, tempPath, -301, cDims); /* Assigns the shared_ptr<>(this, tempPath, -301, dims);  Assigns the shared_ptr<> to an instance variable that is a weak_ptr<> */
```

Comment:
Did you really want to initialize the array to -301?  Doesn't 0 make more sense?